javalib/src/main/scala/java/lang/process/UnixProcessGen2.scala

```diff
@@ -381,11 +381,18 @@ private[process] object UnixProcessGen2 {
       throw new IOException(s"Failed to create process for command: $cmd")
     }
 
+    @tailrec
+    def fork(attempt: Int = 0): Int = {
+      val pid = unistd.fork()
+      if (pid == -1 && errno == EAGAIN && attempt < 10) {
+        // start with 16ms, end with approx 16s
+        Thread.sleep(2 << (attempt + 4))
+        fork(attempt + 1)
+      } else pid
```

Comment:
16 seconds to fork might be way too much. Let's limit to 3 retries. Total of ~250ms backoff should be enough in most cases. If for some reason it would not, it might be better to throw error then hide the underlying issues, especially since that would be difficult to debug